src/client/View.ts

```diff
@@ -0,0 +1,147 @@
+    this.tooltip = document.createElement("div");
+  }
+
+  public setup(mapWidth: number, mapHeight: number) {
+    this.tooltip.id = "tooltip";
+    this.cellWidth = Math.floor(CANVAS_WIDTH / mapWidth);
+    this.cellHeight = Math.floor(CANVAS_HEIGHT / mapHeight);
+
+    const container = document.createElement("div");
+    container.id = "game-container";
+    document.body.appendChild(container);
+    this.stage = new Stage({
+      width: this.cellWidth * mapWidth,
+      height: this.cellHeight * mapHeight,
+      container: "#"+container.id,
+    });
+    this.layer = new Layer();
+    this.stage.add(this.layer);
+
+    document.body.appendChild(this.tooltip);
+  }
+
+  public initField(gameField: (Entity | null)[][]) {
+    gameField.flat(2).filter(Boolean).forEach(entity => {
+      entity = entity as Entity;
+      this.addEntity(entity, entity.x, entity.y);
+    });
+  }
+
+  public addEntity(entity: Entity, x: number, y: number) {
+    const shape: TRect = new Rect({
+      x: x*this.cellWidth,
+      y: y*this.cellHeight,
+      width: this.cellWidth,
+      height: this.cellHeight,
+      fill: this.getEntityColor(entity),
+    });
+
+    if (this.mouseLeaveHandler)
+      shape.on("mouseleave", this.mouseLeaveHandler);
+    if (this.mouseMoveHandler)
+      shape.on("mousemove", this.mouseMoveHandler);
+
+    this.shapes.push(shape);
+    this.layer.add(shape);
+    this.layer.draw();
+  }
+
+  public moveEntity(entity: Entity, [x, y]: [number, number]) {
+    const shape = this.getShape(x, y);
+    const offsetX = (entity.x - x)*this.cellWidth;
+    const offsetY = (entity.y - y)*this.cellHeight;
+
+    shape?.move({ x: offsetX, y: offsetY });
+    this.layer.draw();
+  }
+
+  public removeEntity(x: number, y: number) {
+    const shape = this.getShape(x, y);
+
+    shape?.destroy();
+    this.layer.draw();
+  }
+
+  public updateEntityColor(entity: Entity) {
+    const shape = this.getShape(entity.x, entity.y);
+
+    shape?.fill(this.getEntityColor(entity));
+    this.layer.draw();
+  }
+
+  public coordsToIndexes(offsetX: number, offsetY: number): [number, number] {
+    const x = Math.floor(offsetX / this.cellWidth);
+    const y = Math.floor(offsetY / this.cellHeight);
+
+    return [x, y];
+  }
+
+  public showTooltip(text: string, top: number, left: number) {
+    this.tooltip.innerText = text;
+    this.tooltip.style.setProperty("top", top + "px");
+    this.tooltip.style.setProperty("left", left + "px");
+  }
+
+  public hideTooltip() {
+    this.tooltip.style.removeProperty("top");
+    this.tooltip.style.removeProperty("left");
+  }
+
+  private defineEntity(entity: any) {
+    if (entity._entityType == "Player" && entity.team == "Red") return "Player";
+    if (entity._entityType == "Zombie" || entity?.team == "Green") return "Zombie";
+    return "Unknown";
+  }
+
+  private getShape(x: number, y: number) {
+    return this.shapes.find(e => e.x() == x*this.cellWidth && e.y() == y*this.cellHeight);
```

Comment:
Change `==` to `===`.

Reply:
`TRect.x()` returns number, `x` is a number and `this.cellWidth` is also number, so we should not use strict comparison.